unit-tests/src/org/commcare/CommCareTestApplication.java

```diff
@@ -59,32 +60,38 @@ public CommCareApp getCurrentApp() {
     public PrototypeFactory getPrototypeFactory(Context c) {
         TestUtils.disableSqlOptimizations();
 
-        if (testPrototypeFactor != null) {
-            return testPrototypeFactor;
+        if (testPrototypeFactory != null) {
+            return testPrototypeFactory;
         }
 
-        ArrayList<String> classNames = new ArrayList<>();
         // Sort of hack-y way to get the classfile dirs
-        String baseODK = BuildConfig.BUILD_DIR + "/intermediates/classes/commcare/debug/";
-        String baseJR = BuildConfig.PROJECT_DIR + "/../javarosa/build/classes/main/";
-        String baseCC = BuildConfig.PROJECT_DIR + "/../commcare/build/classes/main/";
-        addExternalizableClassesFromDir(baseODK, classNames);
-        addExternalizableClassesFromDir(baseCC, classNames);
-        addExternalizableClassesFromDir(baseJR, classNames);
         PrefixTree tree = new PrefixTree();
+        getTestPrototypeFactoryClasses();
 
         try {
-            for (String cl : classNames) {
+            for (String cl : factoryClassNames) {
                 tree.addString(cl);
             }
         } catch (Exception e) {
             throw new RuntimeException(e);
         }
 
-        testPrototypeFactor = new AndroidPrototypeFactory(tree);
-        return testPrototypeFactor;
+        testPrototypeFactory = new AndroidPrototypeFactory(tree);
+        return testPrototypeFactory;
     }
 
+    public static List<String> getTestPrototypeFactoryClasses() {
```

Comment:
Confused by the naming of this method in general so an explanation would be helfpul, but it should likely say something more than just "get" since it has side effects from calling `addExternalizableClassesFromDir`